how-we-work/how-to-upgrade-dependencies.md

```diff
@@ -0,0 +1,96 @@
+
+To perform the upgrades you can make use of `pip-upgrader` to figure out which packages have recent updates:
+
+1. Install [pip-upgrader](https://pypi.org/project/pip-upgrader/): `pip install pip-upgrader`.
+
+1. Upgrade the packages based on the report and verify that the application still works (please fully deploy to a `dev` or `rdev` environment to double check everything).
+
+You can also make use of the [Snyk reports](https://app.snyk.io/org/cellxgene) to figure out which specific packages have security vulnerabilities as well as the [`safety` package](#Upgrade-cellxgene-Explorer-desktop-backend). Please note, since the `safety` package only checks currently installed package versions, please create a fresh virtual environment and install the requirements fresh from the `requirements.txt` file before running the command.
+
+Finally, besides the `pip` installed packages, please also take a look at the Dockerfile (in the top level directory `cellxgene/Dockerfile`) to ensure that there are no security vulnerabilities there. Snyk is your friend to figure out if there are and what the remediation process is.
+
+### Upgrading cellxgene Explorer desktop backend
+
+The policy we have to upgrade the backend of cellxgene Explorer _desktop_ is different than any of the other policies. Because many users may have different versions of the dependent `pip` packages already installed on their laptop, and because many users do not use a virtual environment, we need to be _as permissive as possible_ when defining what the allowable versions of each package are.
+
+This means that the allowable versions we set for each package:
+
+1. SHOULD be the minimum version that is functionally compatible and includes no medium or high security risks.
+
+1. MAY exclude specific versions that have medium or high security vulnerabilities if the specification is easier (e.g. `foo>=1.9.0,!=2.0.0`).
+
+To update the `pip` packages for cellxgene Explorer desktop backend, we recommend doing the following steps:
+
+1. Create a clean new virtual environment and install based on the `cellxgene/backend/server/requirements.txt` file. This is necessary because the package in the next step ONLY checks for vulnerabilities against the currently installed packages so you need to ensure that your versions match those in the requirements file.
+
+1. Install [safety](https://pypi.org/project/safety/): `pip install safety`.
+
+1. Manually install the minimum required version of all packages in `cellxgene/backend/server/requirements.txt` and `cellxgene/backend/server/requirements-prepare.txt`.
+
+1. Run `safety`: `python -m safety check --full-report`
+
+1. Update requirements as indicated, goto step 2 again.  Repeat until clean.
+
+1. Check that cellxgene still works :)
```

Comment:
as a future enhancement, you might want to add a git workflow that specifically tests that cellxgene passes unit & smoke tests when the minimum versions are installed.  This enhancement is unrelated to this document, but would catch incompatibilities sooner....